cmd/context/utils.go

```diff
@@ -0,0 +1,215 @@
+		user, err = login.WithToken(ctx, oktetoURL, token)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Infof("authenticating with browser code")
+		user, err = login.WithBrowser(ctx, oktetoURL)
+		if err != nil {
+			return err
+		}
+	}
+
+	if user.New {
+		analytics.TrackSignup(true, user.ID)
+	}
+	log.Infof("authenticated user %s", user.ID)
+
+	if oktetoURL == okteto.CloudURL {
+		log.Success("Logged in as %s", user.ExternalID)
+	} else {
+		log.Success("Logged in as %s @ %s", user.ExternalID, oktetoURL)
+	}
+
+	analytics.TrackLogin(true, user.Name, user.Email, user.ID, user.ExternalID)
+	return nil
+}
+
+func getClusterList() []string {
+	contextList := make([]string, 0)
+	kubeConfigFile := config.GetKubeConfigFile()
+	config, err := okteto.GetKubeConfig(kubeConfigFile)
+	if err != nil {
+		return contextList
+	}
+	for name := range config.Clusters {
+		contextList = append(contextList, name)
+	}
+	return contextList
+}
+
+func selectCluster() (string, error) {
+	clusterList := getClusterListOptions()
+
+	prompt := promptui.Select{
+		Label: "Select the cluster you want to point to:",
+		Items: clusterList,
+		Size:  len(clusterList),
+		Templates: &promptui.SelectTemplates{
+			Label:    "{{ .Name }}",
+			Selected: "{{if .Enable}} ✓  {{ .Name | oktetoblue }}{{else}}{{ .Name | oktetoblue}}{{end}}",
+			Active:   fmt.Sprintf("{{if .Enable}}%s {{ .Name | oktetoblue }}{{else}}{{ .Name | oktetoblue}}{{end}}", promptui.IconSelect),
+			Inactive: "{{if .Enable}}  {{ .Name | oktetoblue}}{{else}}{{ .Name | oktetoblue}}{{end}}",
+			FuncMap:  promptui.FuncMap,
+		},
+	}
+
+	prompt.Templates.FuncMap["oktetoblue"] = log.BlueString
+
+	i, _, err := prompt.RunCursorAt(1, 0)
+	if err != nil {
+		log.Infof("invalid cluster: %s", err)
+		return "", fmt.Errorf("invalid cluster")
+	}
+	if clusterList[i].Name == "Okteto clusters:" || clusterList[i].Name == "Kubernetes clusters:" {
+		log.Infof("invalid cluster")
+		return "", fmt.Errorf("invalid cluster")
+	}
+	return clusterList[i].Name, nil
+}
+
+func getClusterListOptions() []SelectItem {
+	k8sClusters := getClusterList()
+	clusterOptions := []SelectItem{
+		{
+			Name:   "Okteto clusters:",
+			Enable: false,
+		},
+		{
+			Name:   "Okteto cloud",
+			Enable: true,
+		},
+		{
+			Name:   "Okteto enterprise",
+			Enable: true,
+		},
+	}
+	if len(k8sClusters) > 0 {
+		clusterOptions = append(clusterOptions, SelectItem{
+			Name:   "Kubernetes clusters:",
+			Enable: false,
+		})
+	}
+	for _, k8sCluster := range k8sClusters {
+		clusterOptions = append(clusterOptions, SelectItem{
+			Name:   k8sCluster,
+			Enable: true,
+		})
+	}
+	return clusterOptions
+}
+
+func isOptionAnOktetoCluster(option string) bool {
+	return option == "Okteto cloud" || option == "Okteto enterprise" || isURL(option)
+}
+
+func getOktetoClusterUrl(ctx context.Context, option string) string {
+	if option == "Okteto cloud" {
+		return okteto.CloudURL
+	}
+
+	return AskForLoginURL(ctx)
+}
+
+func isURL(okContext string) bool {
```

Comment:
pretty much anything can be a valid URL (e.g "localhost" is a valid URL). If what we need to know is if "this URL belongs to an okteto instance" I think we should do something more robust. maybe there's an 'info' API we can use for this?